### ISSUE TITLE
Unable to view data in discover tab for heartbeat, journalbeat and Functionbeat.

### ISSUE THREAD

**Issue by dikshachauhan-qasource** [elastic/beats#24849]
While doing some installation test for different stand-alone Beats  on ARM Linux support, we have come up with below observations. On following the steps as mentioned in guide, we are still unable to view data in discover tab. 

**Link followed:** https://www.elastic.co/guide/en/beats/libbeat/current/beats-reference.html

**Artifact links used:** 
https://snapshots.elastic.co/7.13.0-8cf5aa00/downloads/beats/heartbeat/heartbeat-7.13.0-SNAPSHOT-linux-arm64.tar.gz
https://snapshots.elastic.co/7.13.0-8cf5aa00/downloads/beats/journalbeat/journalbeat-7.13.0-SNAPSHOT-linux-arm64.tar.gz

1. **Heartbeat observations:**
- Followed steps in guide and made required changes as mentioned.
Yml file: [heartbeat_yml.txt](https://github.com/elastic/beats/files/6228628/heartbeat_yml.txt)
logs file: [heartbeat_logs.txt](https://github.com/elastic/beats/files/6228646/heartbeat_logs.txt)

Screenshot after running below commands:
```
 ./heartbeat setup
sudo chown root heartbeat.yml 
sudo ./heartbeat -e

```
![image](https://user-images.githubusercontent.com/72375798/112981956-826a3180-9179-11eb-96fe-a542c4524708.png)

2. **Journalbeat observations:**
- unable to install journalbeat with provided steps.
yml file: [Uploading journalbeat_yml.txt…]()
logs file: 
[journalbeat_logs.txt](https://github.com/elastic/beats/files/6228992/journalbeat_logs.txt)

Commands used:
```
sudo ./journalbeat -e -c journalbeat.yml
./journalbeat setup
sudo chown root journalbeat.yml 
sudo ./journalbeat -e
```

Screenshot: 
![image](https://user-images.githubusercontent.com/72375798/112988407-6ec2c900-9181-11eb-953c-df4a5fd9ce9d.png)

3. **Functionbeat observations:**

@urso As loction is not specified on guide page, could you please confirm where to update the user permissions for aws with policy content provided on [link](https://www.elastic.co/guide/en/beats/functionbeat/current/iam-permissions.html#iam-permissions-cloudwatch)

yml file: 
[functionbeat_yml.txt](https://github.com/elastic/beats/files/6229444/functionbeat_yml.txt)

**Comment by manishgupta-qasource** [comment]
Reviewed & assigned to @urso

**Comment by urso** [comment]
## Heartbeat

I noticed that kibana URL is not configured. Heartbeat most likely did not setup the index templates, as this requires dashboard setup to be enabled. Can you check the available indices via `/_cat/indices` in the kibana console?

## Journalbeat

The logs look OK. What was the error during setup? Does the index exist (similar to Heartbeat)?

## Functionbeat

@kvch/@ph can you have a look please?

**Comment by kvch** [comment]
@dikshachauhan-qasource How did you create the AWS user? There are a few ways to change permissions detailed here: https://docs.aws.amazon.com/IAM/latest/UserGuide/id_users_change-permissions.html

Let me know if you need more pointers.

**Comment by dikshachauhan-qasource** [comment]
Hi @urso cc @EricDavisX 

We have revalidated the issues reported as per your feedback on 7.13 snapshot build and our observations are as follows:

Further, we need to update guide links with above info as no hint is mentioned there for the user who is attempting it for first time. Let me know if it seems to you right.

### For Heartbeat

We updated the Kibana url in hearbeat.yml and re ran the set up and followed commands, still no data was observed at Discover tab.

**Yml file link**: [heartbeat_yml.txt](https://github.com/elastic/beats/files/6248553/heartbeat_yml.txt)
**Logs:**  [heartbeat.txt](https://github.com/elastic/beats/files/6248628/heartbeat.txt)

**Heartbeat Indices Screenshot for Kibana API console:**
![image](https://user-images.githubusercontent.com/72375798/113398052-9bfabb80-93bb-11eb-9ccc-dadfb7866743.png)

No data found under Kibana folder. Reference screenshot:
![image](https://user-images.githubusercontent.com/72375798/113401083-77eda900-93c0-11eb-9cd6-33fd0d7cacbd.png)

Please let us know if we have misunderstood anything wrong here.

### Journalbeat Indices screenshot from Kibana API Console:
![image](https://user-images.githubusercontent.com/72375798/113400442-6f48a300-93bf-11eb-9421-021a05ab7a29.png)

**Yml file**:  [journalbeat_yml.txt](https://github.com/elastic/beats/files/6248700/journalbeat_yml.txt)

**Logs:** [journalbeat.txt](https://github.com/elastic/beats/files/6248701/journalbeat.txt)

No errors were observed in powershell after running the commands.

### For Functionbeat

No we didn't create any aws user. We were just using account through which we have access to AWS-dev app. 

On running Functionbeat, an error "failed to retrieve aws credentials" showed up. With some research we found, that we need to update aws_access_key and looked for same at location:
![Functionbeaterror](https://user-images.githubusercontent.com/72375798/113402137-2cd49580-93c2-11eb-80c8-d1947406529d.png)

![image](https://user-images.githubusercontent.com/72375798/113404740-4546af00-93c6-11eb-8793-0e5e8a6432eb.png)

Yml file: 
[functionbeat_yml.txt](https://github.com/elastic/beats/files/6248756/functionbeat_yml.txt)

@kvch Please have a look and let us know if we have missed anything.

Thanks
QAS

**Comment by urso** [comment]
## Heartbeat + Journalbeat

The good news is that the indices have been generated and seem to include data. Kibana still is requires index patterns, such that the data can be searched via Kibana discovery. Can you get us a list of the available Kibana index patterns?

**Comment by dikshachauhan-qasource** [comment]
Hi @urso 

We have rechecked the scenario for  Kibana index patterns and couldn't find anything on Index patterns tab.

Please refer below screenshot and let us know if anything is required.

![image](https://user-images.githubusercontent.com/72375798/114016996-167a7e00-9889-11eb-833d-61dd36bfac5a.png)

Thanks
QAS

**Comment by dikshachauhan-qasource** [comment]
Hi @EricDavisX 

We have tried to reproduce these on latest shipped GA 7.13.2 build and reported 03 separate issues for each as per your feedback with details such as guide link and steps used. Links for reported issues are as follows:

- https://github.com/elastic/beats/issues/26366
- https://github.com/elastic/beats/issues/26364
- https://github.com/elastic/beats/issues/26369

For Functionbeat, I'll keep environment ready before our next sync up and we can spend some time together then in debugging.

For now, I'm closing this out.

Thanks
QAS
